VTCareService/src/main/java/edu/vt/cs/vtcare/vtcareservice/api/VTCareResource.java

```diff
@@ -20,4 +22,22 @@ public Provider createProvider(Provider provider) throws Exception {
             throw new Exception("An exception occurred while persisting the provider details.");
         }
     }
+
+    /**
+     *
+     * @param providerid to search for a specific provider
+     * @return the provider
+     * @throws Exception if cascaded from DAO.
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("providers")
+    public List<Provider> getProvider(@QueryParam("providerid") String providerid) throws Exception {
+        try {
+            ProviderService providerService = new ProviderService();
+            return providerid != null ? providerService.getProvider(providerid) : providerService.getProviders();
+        } catch(Exception e) {
+            throw new Exception("An exception occurred while persisting the provider details.");
```

Comment:
Please correct the exception message. We are not persisting the providing, we are fetching a list of providers.